backend/src/main/java/com/htecgroup/skynest/service/impl/UserServiceImpl.java

```diff
@@ -64,14 +64,41 @@ public void sendVerificationEmail(String emailAddress) {
       }
     }
 
-    String token = jwtEmailVerificationUtils.generateJwtEmailVerificationToken(emailAddress);
+    String token = jwtUtils.generateJwtEmailToken(emailAddress);
 
-    String confirmationLink = jwtEmailVerificationUtils.getConfirmationLink() + token;
-    String emailBody = jwtEmailVerificationUtils.buildEmail(emailAddress, confirmationLink);
+    String confirmationLink = jwtUtils.getEmailConfirmationLink() + token;
+    String emailBody = jwtUtils.buildVerificationEmail(emailAddress, confirmationLink);
     Email emailToSend = new Email(emailAddress, SUBJECT_FOR_EMAIL_CONFIRMATION, emailBody, true);
     emailService.send(emailToSend);
   }
 
+  @Override
+  public void sendPasswordResetEmail(String email) {
+    if (!userRepository.existsByEmail(email)) {
+      throw new UserException(UserExceptionType.EMAIL_NOT_IN_USE);
+    }
+
+    String token = jwtUtils.generateJwtEmailToken(email);
+
+    String confirmationLink = jwtUtils.getPasswordResetLink() + token;
+    String emailBody = jwtUtils.buildPasswordResetEmail(email, confirmationLink);
+    Email emailToSend = new Email(email, SUBJECT_FOR_PASSWORD_RESET, emailBody, true);
+    emailService.send(emailToSend);
+  }
+
+  @Override
+  public String resetPassword(String token, String password) {
+    boolean validateToken = jwtUtils.validateJwtToken(token);
```

Comment:
Maybe method validateJwtToken can throw exception if token is not valid and to be void, instead boolean ?

Reply:
That makes sense, will make the method look cleaner as it won't have the if statement and exception throwing. Will change